pep-0497-externals.txt

```diff
@@ -0,0 +1,359 @@
+non-Python dependency types allows. The absence of this prefix implies the the
+package name is a discoverable Python package.
+
+Following the ':' is a comma-separated list of key-operator-value pairs. These
+represent the sufficient trigger conditions to apply the rule. At this stage
+only a very limited set of allowable operators and logics are permitted. The
+goal is to provide just enough support to allow the specification of key
+dependencies for specific environments, not to become a make/build system.
+
+Implementation Concerns
+=======================
+
+Short Version
+-------------
+
+1. Setuptool + pip should display relevant information earlier than now
+2. A mechanism to assist in identifying external package names would be useful
+3. The maintenance burden for (2) should fall on the package management system
+   maintainers, not on pip/setuptools developers
+4. This means a simple pluging approach to mapping from specific resources
+   to relevant external package names and invocation calls
+5. I propose this looks like "pip install vendor_magic darwin"
+6. And that the various O/Ses ship this pre-installed in their distributions
+
+
+Discussion of Implementation Concerns
+-------------------------------------
+
+A common user experience is to attempt the installation of software in a
+basic environment, and for the installation to fail during the compilation
+of a dependency. The user is presented with an error message provided by the
+compiler. Not all users have the ability to understand these messages, and
+a common experience is to spend time searching online for the appopriate
+package which is provided by the vendor.
+
+The base implementation will also only provide information about missing
+headers, libraries or executables. This is a step forward in that the error
+message is provided very early in the install process, rather than potentially
+many minutes in to a large compile job.
+
+A key aspect of the usability of this tool is a mechanism to provide more useful
+information to the user about how to obtain those missing resources.
+The proposal is to provide a layer inside of pip which is capable of providing
+information about appropriate package names. It is surprising that vendors
+do not already provide a good tool for this, given that finding appropriate
+header and library files is a common experience in the general linux ecosystem.
+However, it is also a critical aspect of the user experience, and something
+which package maintainers will need to be able to specify precisely. The
+proposed approach would be for pip to supply the ability to install an
+inference layer, usable by users and vendors alike. Usage would be along
+the lines of
+
+    pip install vendor_magic darwin
+
+This would transform the user messages relating to missing dependencies from
+
+    Could not find libatlas.so
+
+to
+
+    Could not find libatlas.so. Run 'port install scipy +atlas'
```

Comment:
I think this is crossing layers here - I'd expect the error to describe how to install libatlas.so, not how to install scipy which is just one user of it.